### DISCUSSION TITLE
Kafka UI OAUTHBEARER broker settings

### DISCUSSION THREAD

**Discussion by xdeveloper** [kafbat/kafka-ui#433] [Q&A]
**Hi, Team!**

I have couple of parameters but do not understand how to use them in Kafka UI.

"security.protocol"                   = "SASL_SSL"
"sasl.mechanism"                      = "OAUTHBEARER"
"sasl.oauthbearer.token.endpoint.url" = "https://blablabla.oktapreview.com/oauth2/ausud5wgoxiBdIXXX0h7/v1/token"
"sasl.login.callback.handler.class"   = "org.apache.kafka.common.security.oauthbearer.secured.OAuthBearerLoginCallbackHandler"
"sasl.jaas.config"                    = "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required clientId="client_id" clientSecret="client_secret" scope="user";"

How to do that using UI or by editing some conf files?

Please help!

**Comment by Haarolean** [comment]
You can use our UI setup wizard ([docs link](https://ui.docs.kafbat.io/configuration/configuration-wizard), where you can choose your auth method (oauthbearer included)

**Reply by xdeveloper** [reply]
@Haarolean no idea how to put my parameters into Wizard.

**Reply by xdeveloper** [reply]
![image](https://github.com/kafbat/kafka-ui/assets/988712/8af37650-0d6c-459a-a4d1-e9d8e03eedbe)

**Reply by xdeveloper** [reply]
I have: sasl.oauthbearer.token.endpoint.url, sasl.login.callback.handler.class, sasl.jaas.config

**Reply by Haarolean** [reply]
https://kafka.apache.org/20/javadoc/org/apache/kafka/common/security/oauthbearer/OAuthBearerLoginModule.html
```
Here is a typical, basic JAAS configuration for a client leveraging unsecured SASL/OAUTHBEARER authentication:

 KafkaClient {
      org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule Required
      unsecuredLoginStringClaim_sub="thePrincipalName";
 };
```

**Comment by xdeveloper** [comment]
Is there any chance to use OAUTH for Schema Registry?

**Reply by Haarolean** [reply]
I doubt SR supports OAuth. How are you supposed to authenticate as a consumer/producer in this case?

**Comment by xdeveloper** [comment]
I use bearer.auth.credentials.source: "SASL_OAUTHBEARER_INHERIT" for Schema
Registry configuration.

I inherits auth data from kafka client (do not know how but it works on
real project)

чт, 1 серп. 2024 р. о 20:22 Roman Zabaluev ***@***.***> пише:

**Reply by Li-Dongmeng** [reply]
I have encountered the same problem now. How did you solve it at that time. Please provide a more detailed method

**Comment by MariaIRubio** [comment]
Hi! I am having problems to configure my broker in the UI with SASL_SSL y OAUTHBEARER mechanism. 
How should the sasl.jaas.config be?

Something like this?
```
org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required  oauth.client.id="clientid" oauth.client.secret="clientsecret" oauth.token.endpoint.uri="issuer/auth/realms/realm/protocol/openid-connect/token"; sasl.oauthbearer.sub.claim.name="preferred_username"; sasl.client.callback.handler.class="io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler";
```
like this?

```
KafkaClient {
	org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule Required 
	usecuredLoginStringClaim_sub="preferred_name";
};

```

like this?

```
KafkaClient{
org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule required  
oauth.client.id="clientid" 
oauth.client.secret="clientsecret" 
oauth.token.endpoint.uri="issuer/auth/realms/realm/protocol/openid-connect/token";
sasl.oauthbearer.sub.claim.name="preferred_username"; 
sasl.client.callback.handler.class="io.strimzi.kafka.oauth.client.JaasClientOauthLoginCallbackHandler";
};
```

something totally different?

Thank you very much in advance

**Comment by Haarolean** [comment]
https://github.com/kafbat/kafka-ui/discussions/857#discussioncomment-12297978

**Comment by credmond** [comment]
@Haarolean: sasl.oauthbearer.token.endpoint.url is not a JAAS config setting, it's a top-level setting.

None of the "sasl." properties are, I believe. Kafbat, when adding a new cluster in the UI, doesn't allow you to specify sasl.oauthbearer.token.endpoint.url. Not sure how more people haven't reported that.
